docs/okeps/okep-5085-localnet-api.md

```diff
@@ -0,0 +1,825 @@
+The workloads networking should not break, and workloads should be able to communicate with each other. 
+
+The proposed solution suggest to allow users mutate the following settings for localnet topology networks: 
+`VLAN`, `MTU`, `PhysicalNetworkName` and `ExcludeSubnets`.
+
+See below description on how each setting change may or may not affect the workloads:
+- `VLAN`, `PhysicalNetworkName`:
+  Changing `VLAN` or `PhysicalNetworkName` should not introduce network disruption to connected workloads.
+  In edge cases there could be some packet drops, but no TCP connection breakage is expected.
+- `MTU`
+  As of today the CNI configure the MTU on the pod interface. The CNI is executed on pod create or deletion.
+  In order for MTU change take effect the workloads should be re-created in order to have the CNI apply the new MTU settings.
+
+  In a scenario some pods configured with new MTU settings, and other pods configured with old MTU settings.
+  - Pods should be able to communicate with each other.
+  - If the external network interface MTU is higher than old MTU settings, there might be network disruption causing packet fragmentation.
+
+- `ExcludeSubnets`
+  ExcludeSubnet can be configured when `subnets` is configured, in this case OVN-Kubernetes will assigned pod IPs to pods from the specified
+  value in `subnets`.
+  When `ExcludeSubnet` is set, OVN-Kubernetes exclude the specified IP address from the pool and wont assign IPs in this range to pods.
+
+  Adding or removing an exclude-subnet should not introduce network disruption whatsoever, as this setting effects OVN-Kubernetes IPAM service
+  and not the pods network-stack directly.
+
+  In a scenario the external network introduce new "service" whose IP address now considered reserved, e.g.: 10.0.0.10.
+  In case there the address 10.0.0.10 already assigned a pod, both the "service" and the pod may suffer from network disruption  
+  due to IP address conflict.
+  Once the pod with the assigned IP is re-created, it will get different, non excluded, IP address from the pool, and the IP conflict
+  should be resolved.
+
+  Mitigation for this can be providing indication about this state - excludeSubnet cannot be set because IP in its range already
+  been assigned to pods.
+
+
+Once all the connected workloads consume the new network-config, no network disruption should occur as all connected workloads
+network settings are aligned, using the latest network-config.
+
+## OVN Kubernetes Version Skew
+
+## Alternatives
+
+### Indicating pod configured with outdated network-config
+Following discussions about the proposed solution for indicating pod configured with outdated network-config,
+some alternatives and improvements were suggested.
+The proposed solution serves as a baseline for most suggested alternatives.
+Most alternative solutions are additions serving as an additional matter for providing similar indication, as the proposed one.
+
+#### 1. Emit events on pods 
```

Comment:
IMO (and i also think @jcaamano  agrees) this is a requirement and not optional for this API change. The event is a much more kubernetes way to know to see that something is wrong with a pod when a user does kubectl describe.

Reply:
do you mean together with the annotation or instead of the annotation?

Reply:
Moved emitting event as part of the proposed solution.